pkg/activator/net/throttler.go

```diff
@@ -592,13 +592,25 @@ func (rt *revisionThrottler) handlePubEpsUpdate(eps *corev1.Endpoints, selfIP st
 	epSet, _ := endpointsToDests(eps, rt.protocol)
 	// We are using List to have the IP addresses sorted for consistent results.
 	epsL := epSet.List()
-	atomic.StoreInt32(&rt.numActivators, int32(len(epsL)))
-	atomic.StoreInt32(&rt.activatorIndex, int32(inferIndex(epsL, selfIP)))
-	// Note that if the revision is served directly or this activator is not
-	// part of the subset it will be `-1/X`. And that's OK, since this activator
-	// should not be receiving requests for the revision.
-	rt.logger.Infof("This activator index is %d/%d", rt.activatorIndex, rt.numActivators)
-	rt.updateCapacity(rt.backendCount)
+	na, ai := int32(len(epsL)), int32(inferIndex(epsL, selfIP))
+
+	// Only update the values if they have changed and are meaningful.
+	// If ai == -1, means this activator is not part of the gang (or SKS
+	// is in serve mode), so the change should not matter, since this activator
+	// should not be receiving the requests anyway.
+	// NB: the line below is the only place where the number of activators
+	// or activator index are changed. Thus we can check the values without
+	// atomic ops here.
```

Comment:
Are you sure about that? My spidey senses tingle a little here. If you ask me, I'd put the Load statements in here for good measure and remove 3 lines of comment for it.